### PR TITLE
TES-277: Packer Automation

### PR DESCRIPTION
**Pull request by mihailradkov**
## Description

Overhauled the variables and scripts structure.

## Changes

- Added proper prefixes to the variable names in variables.pck.hcl
- Grouped variables by usage in variables.pck.hcl
- Added new variables in variables.pck.hcl
  - `tags` - Additional tags to apply to the build resources
  - `build_temp_resource_group_name` - Name of the temporary resource group created by Packer during the build
  - `build_resource_group_name` - Name of a resource group to be used by Packer during the build
  - `gallery_subscription_id` - ID of the subscription where the Shared Image Gallery is located
- Updated create_image_definition.sh to just create the definition and to skip invoking `packer build`
- Removed the ARM configuration for building images

## Related Issues

TES-277

### CODE REVIEW

File: variables.pkr.hcl [hunk 1]
```diff
@@ -24,93 +38,133 @@ variable client_secret {
...
+  description = "A list of regions to replicate the image version in"
+  type        = list(string)
+  default     = []
 }
 
-variable image_replica_count {
+variable gallery_image_replica_count {
   description = "The number of replicas of the Image Version to be created per region. Replica count must be between 1 and 100"
   type        = number
   default     = 1
```
Comment (viktor-ribchev) [inline variables.pkr.hcl:163]: better change this to  3

File: variables.pkr.hcl [hunk 2]
```diff
@@ -24,93 +38,133 @@ variable client_secret {
...
 
-# Generic variables
+#
+# GraphDB variables
+#
 
-variable primary_location {
-  description = "Azure datacenter in which your VM will build"
+variable graphdb_version {
+  description = "Version of GraphDB which will be installed and packaged as VM"
```
Comment (viktor-ribchev) [inline variables.pkr.hcl]: as VM image

File: variables.pkr.hcl [hunk 3]
```diff
@@ -24,93 +38,133 @@ variable client_secret {
...
-  type        = string
-}
+#
+# Build variables
+#
 
-variable gdb_version {
-  description = "Version of GraphDB which will be installed"
+variable build_location {
+  description = "The Azure region where the VM will be built with Packer"
```
Comment (viktor-ribchev) [inline variables.pkr.hcl]: ...where the VM image ...

**Review by viktor-ribchev — APPROVED** [review]

File: variables.pkr.hcl (hunk 3 above)
Comment (mihailradkov) [reply variables.pkr.hcl]: Done

File: variables.pkr.hcl (hunk 2 above)
Comment (mihailradkov) [reply variables.pkr.hcl]: Done

File: variables.pkr.hcl (hunk 1 above)
Comment (mihailradkov) [reply variables.pkr.hcl:163]: I override this in the release pipeline as needed.

File: files/install_graphdb.sh [hunk 4]
```diff
@@ -112,14 +113,18 @@ echo 'fs.file-max = 262144' | tee -a /etc/sysctl.conf
...
+echo "keepalive and max file size are configured"
+
 echo "##################################################"
 echo "#    Setting Azure Application Insights agent    #"
 echo "##################################################"
 
-wget -P /opt/graphdb/ https://github.com/microsoft/ApplicationInsights-Java/releases/download/3.4.19/applicationinsights-agent-3.4.19.jar
+curl -fsSL -O --output-dir /opt/graphdb/ https://github.com/microsoft/ApplicationInsights-Java/releases/download/3.4.19/applicationinsights-agent-3.4.19.jar
 ln -s /opt/graphdb/applicationinsights-agent-3.4.19.jar /opt/graphdb/applicationinsights-agent.jar
 chown graphdb:graphdb /opt/graphdb/applicationinsights-agent.jar
```
Comment (mihailradkov) [inline files/install_graphdb.sh]: This does not fully fix the permissions
```
-rw-r--r-- 1 graphdb graphdb  39M Feb  7 13:28 applicationinsights-agent-3.4.19.jar
lrwxrwxrwx 1 root    root      49 Feb  7 13:28 applicationinsights-agent.jar -> /opt/graphdb/applicationinsights-agent-3.4.19.jar
```

Comment (mihailradkov) [reply files/install_graphdb.sh]: Updated

**Pull request merged**